core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -301,6 +301,9 @@
 
     private static final boolean USE_JDK_DNS_RESOLVER = getBoolean("useJdkDnsResolver", false);
 
+    private static final boolean UNSAFE_DISABLE_HEADER_VALIDATION =
+            getBoolean("unsafeDisableHeaderValidation", false);
```

Comment:
How about `headerValidationEnabled` with `true` default?

Reply:
Sure - went with `validateHeaders`, which is somewhat like `useEpoll` and `dumpSslInfo`